src/graphql.rs

```diff
@@ -255,14 +262,23 @@ impl Query {
     }
 
     #[instrument(skip(self, ctx))]
-    async fn configuration(
+    async fn configurations(
         &self,
         ctx: &Context<'_>,
-        beamline: String,
-    ) -> async_graphql::Result<BeamlineConfiguration> {
+        beamline_filter: Option<String>,
+    ) -> async_graphql::Result<Vec<BeamlineConfiguration>> {
         let db = ctx.data::<SqliteScanPathService>()?;
-        trace!("Getting config for {beamline:?}");
-        Ok(db.current_configuration(&beamline).await?)
+        match beamline_filter {
+            Some(filter) => {
+                trace!("Getting configs matching {filter:?}");
+                let singleton = db.current_configuration(&filter).await?;
```

Comment:
I've done this but I'm on the fence about it. The DB service returns an error if it can't find the beamline because the other queries like to fail when there's no configuration. It now seems a bit inconsistent. This query basically returns an `Option` and everything else returns a `Result`.